datacube_stats/statistics.py

```diff
@@ -314,6 +314,37 @@ def measurements(self, input_measurements):
                 index_measurements + text_measurements)
 
 
+class Percentile(PerBandIndexStat):
```

Comment:
Could we get a docstring here please, describing the argument `q`, and what outputs using this `Stat` will produce. Understandable at a user level.

I think it would be good to have a command line option which lists the available Stats and how to use them. Similar to `pytest -q --fixtures`.